### PR TITLE
feat/movie list

### PR DESCRIPTION
**Pull request by abwizeline**
#### What does this PR do?
- Showing movie list in main screen

#### How should this be manually tested?
- Pull changes and open index.html

#### What are the relevant issues?
- https://github.com/trqhien/progressive-movie-web-app/issues/12

### CODE REVIEW

File: scripts/fetch.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+var api_key = "fce9002a14d2a9df6121f300d03601c7";
+var api_movie = "https://api.themoviedb.org/3/movie/top_rated?api_key=";
+var api_movie_now = "https://api.themoviedb.org/3/movie/top_rated?api_key=";
+var api_movie_up = "https://api.themoviedb.org/3/movie/top_rated?api_key=";
+var api_movie_pop = "https://api.themoviedb.org/3/movie/top_rated?api_key=";
```
Comment (trqhien) [inline scripts/fetch.js]: I think you can create a `base_url = 'https://api.themoviedb.org/3'` and append appropriate endpoints depending on the API call that we are using

File: scripts/fetch.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+var ApiTypeUrl= {
+    "top": api_movie,
```
Comment (trqhien) [inline scripts/fetch.js]: may be change the name to be more descriptive?

File: scripts/fetch.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    "now": api_movie_now,
```
Comment (trqhien) [inline scripts/fetch.js]: you don't really need the `""`

**Comment by trqhien** [comment]
Please resolve conflicts

File: index.html [hunk 4]
```diff
@@ -49,14 +53,57 @@
   </header>
 
   <main>
-      <h1>Resposive Movie App</h1>
+      <section>
+          <ul id="movies">
```
[the remark this answers is not in the document]
Comment (abwizeline) [reply index.html]: +

File: index.html [hunk 5]
```diff
@@ -49,14 +53,57 @@
...
+        </section>
+
+        <footer>
+            <h1>Footer</h1>
+          </footer>
   </main>
+</body>
+</html>
+
+<script src="scripts/fetch.js" type="text/javascript"></script>
```
[the remark this answers is not in the document]
Comment (abwizeline) [reply index.html]: +

File: index.html [hunk 6]
```diff
@@ -49,14 +53,57 @@
...
-  </footer>
+    for (var i = 0; i < result.length; i++) {
+      let movie = result[i];
+      output += '<li class="snippet">';
+      output += '<div class="flex__snippet">'
+      output += '<picture class="snippet__thumbnail">';
+      output += `<source srcset="${movie.smallPoster()}" media="(max-width: 500px)">`;
+      output += `<source srcset="${movie.poster()}" media="(max-width: 800px)">`;
+      output += `<source srcset="${movie.largePoster()}" media="(min-width: 800px)">`;
+      output += `<img class="snippet__thumbnail_img" src="${movie.poster()}" alt="">`;
```
[the remark this answers is not in the document]
Comment (abwizeline) [reply index.html]: +

File: index.html [hunk 7]
```diff
@@ -49,14 +53,57 @@
...
+      output += '</div>'
+      output += '</div>'
+      output += '</li>';
+    }
 
-  <script>
+    document.getElementById('movies').innerHTML = output;
+  }
+
+  function onError(error) {
```
[the remark this answers is not in the document]
Comment (abwizeline) [reply index.html]: It's standard Promise behaviour. We have to keep it to show error during fetching data. Currently it's not showing in UI but in logs

File: scripts/fetch.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
+var api_key = "fce9002a14d2a9df6121f300d03601c7";
```
[the remark this answers is not in the document]
Comment (abwizeline) [reply scripts/fetch.js]: +

File: scripts/fetch.js (hunk 1 above)
Comment (abwizeline) [reply scripts/fetch.js]: +

File: scripts/fetch.js (hunk 3 above)
Comment (abwizeline) [reply scripts/fetch.js]: +

File: scripts/fetch.js (hunk 2 above)
Comment (abwizeline) [reply scripts/fetch.js]: +

File: index.html [hunk 9]
```diff
@@ -49,14 +53,57 @@
   </header>
 
   <main>
-      <h1>Resposive Movie App</h1>
+      <section>
```
[the remark this answers is not in the document]
Comment (abwizeline) [reply index.html]: @trqhien Can't get what you mean here

[the remark this answers is not in the document]
Comment (trqhien) [reply index.html]: Could be rendering issue from github. The indentation looks good on the VSCode 👍

[the remark this answers is not in the document]
Comment (trqhien) [reply index.html]: I've seen you have fixed that in your latest PR 👍. So it's all good

File: scripts/fetch.js [hunk 10]
```diff
@@ -1,15 +1,16 @@
...
-    "now": api_movie_now,
-    "pop": api_movie_pop,
-    "up": api_movie_up
-    }
+const api_key = "fce9002a14d2a9df6121f300d03601c7";
+const base_url = "https://api.themoviedb.org/3/movie";
+var api_movie = base_url + "/top_rated?api_key=";
+var api_movie_now = base_url + "/top_rated?api_key=";
+var api_movie_up = base_url + "/top_rated?api_key=";
+var api_movie_pop = base_url + "/top_rated?api_key=";
```
Comment (trqhien) [inline scripts/fetch.js]: There're still a lot of duplicated code. But let's refactor that in another PR

File: scripts/fetch.js [hunk 11]
```diff
@@ -1,15 +1,16 @@
...
+
+const TYPE_MOVIEURL= {
+    top: api_movie,
+    now: api_movie_now,
+    pop: api_movie_pop,
+    up: api_movie_up
```
Comment (trqhien) [inline scripts/fetch.js]: I mean change the key name to something more descriptive. I wouldn't understand what `top` or `now` means, but maybe `topRated`, `nowPlaying`

We can merge it for now. Let's refactor it in another PR which @khiemle  is working on

**Pull request merged**